statsmodels/stats/oneway.py

```diff
@@ -0,0 +1,377 @@
+    """effect size corresponding to Cohen's f^2 = nc / nobs for oneway anova
+
+    Parameters
+    ----------
+    means: array_like
+        Mean of samples to be compared
+    vars_ : float or array_like
+        Residual (within) variance of each sample or pooled
+        If var_ is scalar, then it is interpreted as pooled variance that is
+        the same for all samples, ``use_var`` will be ignored.
+        Otherwise, the variances are used depending on the ``use_var`` keyword.
+    nobs : int or array_like
+        Number of observations for the samples.
+        If nobs is scalar, then it is assumed that all samples have the same
+        number ``nobs`` of observation, i.e. a balanced sample case.
+        Otherwise, statistics will be weighted corresponding to nobs.
+    use_var : {"unequal", "equal"}
+        If ``use_var`` is "unequal", then the variances can differe across
+        samples and the effect size for Welch anova will be computed.
+    ddof_between : int
+        Degrees of freedom correction for the weighted between sum of squares.
+        The denominator is ``nobs_total - ddof_between``
+        This can be used to match differences across reference literature.
+
+    Returns
+    -------
+    f2 : float
+        Effect size corresponding to squared Cohen's f, which is also equal
+        to the noncentrality divided by total number of observations.
+
+    Notes
+    -----
+    This currently handles the following 2 cases for oneway anova
+
+    - balanced sample with homoscedastic variances
+    - samples with different number of observations and  with homoscedastic
+      variances
+    - samples with different number of observations and  with heteroscedastic
+      variances. This corresponds to Welch anova
+
+    Status: experimental
+    This function will be changed to support additional cases like
+    Brown-Forsythe anova.
+    We might add additional returns, if those are needed to support power
+    and sample size applications.
+
+
+    """
+    # the code here is largely a copy of onway_generic
+
+    means = np.asarray(means)
+    n_groups = means.shape[0]
+
+    if np.size(nobs) == 1:
+        nobs = np.ones(n_groups) * nobs
+
+    nobs_t = nobs.sum()
+
+    if use_var == "equal":
+        if np.size(vars_) == 1:
+            var_resid = vars_
+        else:
+            vars_ = np.asarray(vars_)
+            var_resid = ((nobs - 1) * vars_).sum() / (nobs_t - n_groups)
+
+        vars_ = var_resid  # scalar, if broadcasting works
+
+    weights = nobs / vars_
+
+    w_total = weights.sum()
+    w_rel = weights / w_total
+    # meanw_t = (weights * means).sum() / w_total
+    meanw_t = w_rel @ means
+
+    # f2 = np.dot(weights, (means - meanw_t)**2) / (n_groups - ddof_between)
+    f2 = np.dot(weights, (means - meanw_t)**2) / (nobs_t - ddof_between)
+
+    # Not sure if I need it in this function
+    compute_df = False
+    welch_correction = False
+    if compute_df:
```

Comment:
no test coverage for this part
delete, and copy paste from anova_generic if we need it later